### ISSUE TITLE
how to use youtube kids on Smart youtube

### ISSUE THREAD

**Issue by bibinvargheset** [yuliskov/SmartTubeNext#1215]
I use two accounts on my smart tv how to set youtube kids on the second account?

**Comment by realkot** [comment]
Same question. Account is added as "empty" and after STB restarts kid account is removed.

**Comment by sneko** [comment]
Did you find a solution @bibinvargheset @realkot ?

cc @yuliskov

**Comment by yuliskov** [comment]
@sneko it's a shame to admit, but I need a guide to set up a kids' account for further testing.

**Comment by sneko** [comment]
I'm new to this field so won't help that much but from what I understand there is 2 possibilities:

1. Either you have your adult Youtube account, and you can create virtual profiles for your kids (name, age...) so they can go on https://www.youtubekids.com/ or the Youtube Kids app (mobile/TV). From there they can only content for their age.
![image](https://user-images.githubusercontent.com/8195958/188994481-abca0c23-a740-4436-833a-40461cb26096.png)

2. For older kids, you can create for them a Google/YT account but when defining their birthdate if will detect if a parent account has to be linked, if so you have control over their accounts and you can define in Youtube settings the level of "aged content" they can see (Explore / Explore More / Almost all Youtube).

So conclusion for me: (2) will restrict videos based on AI (I guess, or manual feedbacks), whereas (1) is a whitelist of videos for specific age.

In both cases ads are still in the game... (https://support.google.com/youtubekids/answer/6130541?hl=en) which sounds terrible since Google makes enough money to not show this to kids :/ (but that's my point of view...).

I guess (2) would work in SmartTubeNext since that's dedicated accounts, but I don't know for (1)... I guess STN should manage those kind of virtual kid profiles and add a `<select>` if any virtual profile exists.

I'm just exploring so I cannot have a strong idea about merging this in the current tool (UI/UX).

**Comment by yuliskov** [comment]
@sneko got it. Thanks for the help.

**Comment by CarrnellTech** [comment]
I would like to add to this. Please take the time to understand my frustration and my search for the perfect solution.

This comes from my experience of running a tech business, helping so many others with this issue, and from raising 3 of my own kids.

YouTube Kids' protocols do not provide enough. It is just enough to get by, but, when a parent wants to participate in their internet journey by helping them along and guiding their experience into the amazing side of YouTube, its impossible. Let me rephrase, its not impossible, just extremely out of reach for the general public (and those who don't want to pay others for something that YouTube should have already figured out a long, long time ago.)

The essential problem is this: **We as parents do NOT get the ability to perfectly curate YouTube.** I say curate because it is close to impossible to manually block each and every channel manually. At that point its a full time job, for the growth of the child.

Yes, YouTube Kids already curates content in their list of Collections, however, that is not perfect in two ways.
1. **It has been known in several instances in the past for bad acting channels to slip through.** Although, thats not much of problem as it will eventually be removed anyway.
2. **You can never add channels that you have already deemed safe for you child**. I can already loose count on how many channels that my child would ABSOLUTLY love to watch on their own, but, YouTube has deemed it not "made for kids." So essentially you are stuck to Collections on that really disappointing app.

There is a function on YouTube Kids that gets you VERY close to what our needs are. "Approve the content yourself." However, if the channel / video is not marked "made for kids" it will not listed, even if you manually marked it approved. (I will admit, I have not used the YouTube Kids app in over a year, however, I have suspicions that this has not changed. I also have problems with their UI design for children (to keep it brief, I believe more trust in children using the adult UI is warranted and designing a UI for children is a waste of time and resources. They eventually need to be educated on how to use it eventually anyway.)

The plot thickens with YouTube creators as well. Creators are faced with a dilemma, I'm not too keen on the exact reasons however, but, it is VERY common to see really awesome creators avoid "made for kids." Think about it, when have you ever seen a child (able to be present and not mindless in the digest of content, most likely in their teens) 9-12 section of YouTube Kids watching a recent relevant Science documentary. No I'm not talking about Discovery or highly partnered/produced channel. I suspect they are paid for just listing the video. Their are plenty of independent channels out there that are worthy of my children's unsupervised watch time.

--------

TLDR; YouTube Kids does not work for the involved parent. YouTube Kids only works as a distraction app with minor education on the side.

--------

I absolutely love this app. **Do you know how close you are to providing my, and many other parents, exact needs!?!**

I hope this is simple, if not, please forgive me as I am ignorant to the methods of programming Android TV Apps.

First, disable suggestions on the bottom of videos, then, set auto play to playlist (or currently selected list of videos like Subscriptions, Channels, etc).

All you would have to do is add a couple checkboxes disabling menu items such as Home, Gaming, News, Music, and Settings. Then put a function in there to add a Parental PIN! Which locks or hides those sections from the child! The account will be subscribed to what is already deemed approved for my child.

You may have to add a checkbox to lock auto play settings, to prevent it from playing a video not subscribed to, if that does happen. And lastly a checkbox to disable video descriptions to prevent clicking on videos off channel.

This situation would be perfect for SO many of my business's clients and for our household! (I only hope that one day this app, or, one of similar needs could be met on an Android Tablet.)

**Comment by yuliskov** [comment]
@CarrnellTech almost all the features you're talking about has been already been implemented. You can hide Home, Gaming, News, Music, and Settings. You can even hide the suggestions.

**Comment by CarrnellTech** [comment]
Oh excellent! I guess I haven't seen those options since I have configured those TVs so long ago. I guess my only feature suggestion then is to add a PIN to prevent settings changes!

**Comment by CarrnellTech** [comment]
After quickly going through the settings I have found the checkboxes to remove those menu items. I apologize for assuming they were not in yet. You have no idea how much that has helped already! My youngest will be ecstatic to be unsurprised!

So... I guest this boils down to a couple of smaller features:

- PIN protecting settings
- A checkbox or method to lock autoplay settings.
- A checkbox to disable the descriptions button.

**Comment by yuliskov** [comment]
@CarrnellTech you can hide any player's button via settings too.

**Comment by CarrnellTech** [comment]
.... I just found it. Haha. Awesome.

**Comment by yuliskov** [comment]
@CarrnellTech if you're struggling to find some settings then let me know.
Currently missing feature:
1) PIN-protecting settings
2) Option to lock autoplay settings.
3) Option to hide search button

**Comment by CarrnellTech** [comment]
@yuliskov Ah, good call on the search button. Didn't think about that one. But, I'm good. I'm just impatient and dumb for not thoroughly combing through the settings.

**Comment by yuliskov** [comment]
@CarrnellTech 

16.38 beta:
- Added the ability to protect the settings with a password. Settings/General/Misc/Protect with a password
- Added ability to hide search button (Settings/General/Top bar buttons)

Seems, that's all that you've asked for.

**Comment by CarrnellTech** [comment]
@yuliskov You have already added it?!? My goodness that was fast. I GREATLY appreciate your time good sir. The very next donation will be from me. I'll donate what I can! The amount I am donating will pale in comparison to my gratitude.

**Comment by yuliskov** [comment]
@CarrnellTech thanks for the generous donation, friend. It's a vital in my situation (Dnipro, Ukraine). Next time, I'll add an option to quickly toggle Child mode (all needed settings will applied automatically).

**Comment by yuliskov** [comment]
@CarrnellTech please give feedback on this feature.

16.39 beta:
- Added Child mode. Settings/General/Misc/Child mode. In this mode, the user can't use search or see any suggested content. The settings will be under the password.

![screen](https://user-images.githubusercontent.com/5897464/200203637-906119b4-ec66-4275-8c25-6ab5b2cb59be.png)

**Comment by CarrnellTech** [comment]
Understood, will report back soon.

**Comment by CarrnellTech** [comment]
@yuliskov Here are my notes on the features good sir:

Signed into one of the child accounts and immediately turned on child mode.

I am also assuming that after child mode is enabled, that settings that I manually re-enable/disable will stay as well.

---
Settings I changed to suite my needs and their reasons why:

- **Enabled Channels Section**
  - I imagine that the subscriptions list may grow to be VERY large for some parents, so, Pin/Unpinning channels could take some time for new setups or resetting up. They have a dedicated child account that we use for subscriptions and I assume channels only shows those subscriptions.
- **Enabled Account button at the top**
  - In my situation I have 3 children that have slightly different tastes. For example, the stereo typical Barbies for girls and Hotwheels for boys. Where one of my kids wont care to see a list of channels the other likes, etc. So being able to switch, add, or remove accounts quickly would be useful. However, I do recognize that this could be abused and may be different for some parents.
- **Enabled Player button Seek Interval**
  - If they want to change the seeking interval for funny or playful situations, I do not see any harm in allowing it.
- **Enabled Player button Playback Queue**
  - This looked like a quick history of sorts. I see no disadvantage in this either.
- **Enabled Player button Screen Off**
  - Situations where they are subscribed to music channels and want the Live video to play but not burn in the screen is also welcome in preserving the life of their display.
- **Enabled Player button Video Zoom**
  - If they want to change the video zoom for funny or playful situations, I do not see any harm in allowing it.
- **Enabled Player button Next Video**
  - If they are playing a pined playlist, or, watching an entire channel, I see no disadvantage in allowing playback controls.
- **Enabled Player button Previous Video**
  - If they are playing a pined playlist, or, watching an entire channel, I see no disadvantage in allowing playback controls.
- **Disabled Player button Chat (On the youngest TV)**
  - This will be different, I imagine, per parent. My oldest children are already adept at filtering out the comments, text, and speech of random internet people. However, my youngest is still learning. So this is disabled only for my youngest.
- **Temporarily Enabled Player button Playback Mode and set it to Play only playlist videos continuously.**
  - Being able to continuously play a section of videos without intervention is welcome. Under my limited testing I have found that the "Play only playlist" keeps it within the confounds of the channel or playlist they were watching. So having it on is not bad. However, I have disabled the ability to change it due to my limited testing of the feature and my desire to limit it going to a different unrelated video.

Thanks again so much for this. Its amazing!

**Comment by CarrnellTech** [comment]
@yuliskov You are quite welcome! Happy to help!

**Comment by yuliskov** [comment]
@CarrnellTech, your propositions have been added to beta 16.44

Direct link to the apk: [kutt.it/stn_beta](https://kutt.it/stn_beta)

**Comment by CarrnellTech** [comment]
Most excellent! Tested the setup process and looking good!

**Comment by freakintoddles2** [comment]
@yuliskov any chance you would consider adding a setting to disable showing trending searches? I don't want my kids seeing or searching for trending searches given the type of content they generally contain.

**Comment by yuliskov** [comment]
@freakintoddles2 you're talking about hiding trends inside Home section? I'm I right?

**Comment by freakintoddles2** [comment]
> @freakintoddles2 you're talking about hiding trends inside Home section? I'm I right?

Hi, I already have the home section hidden, I am talking about when you go to the search bar, below the search bar several suggested trending searches show up before you start typing.

**Comment by yuliskov** [comment]
@freakintoddles2 got it. I'll such an option to the Settings/General/Misc

**Comment by freakintoddles2** [comment]
@yuliskov thank you so much!

**Comment by yuliskov** [comment]
@freakintoddles2 I need your help. Should this option hide previous searches that the user did?

**Comment by freakintoddles2** [comment]
@yuliskov no I think previous searches the user did are fine, it's just the trending searches from the general public that I am suggesting are hidden.
Currently if I go to the Search bar, before I start typing anything in the box, below the search box it shows little pills with trending searches mostly related to the poor state of the world and related news headlines. That is what I would like to have a way to remove. 
So the setting could be called "Disable trending searches" or similar. If disabled, it wouldn't show trending searches from other users anymore.

**Comment by eschweighofer** [comment]
Hello @yuliskov, any chance to add a master password? Either when the app starts or to add a password when switching profiles. As our kids are younger they are not yet allowed to use their account, so we have a setup with 2 adult accounts in smarttube and 1 is used as child, second for us parents so we need to prohibit switching to adult account. I would see 2 options:

Option 1: master password (PIN) when launching the app - here we could have stable + beta on the box and we would enable the master password in the release with parents account.

Option 2: my preferable, but more complex. Add an option to setup a password (PIN) when switching accounts (or option to set the password/PIN) per account). Here also the child mode would need to be per account.

Do you see it feasible? 

Thanks

**Comment by yuliskov** [comment]
@eschweighofer I'll add implement master password fisrt. Then we'll see.

**Comment by eschweighofer** [comment]
@yuliskov, I fully understand and for me, it will be enough, that option 2 is very complex, maybe once you decide to change the app to have settings per account in general... THANKS!

**Comment by freakintoddles2** [comment]
@yuliskov I see the option in Settings / Search / Misc called 'Show trending searches' but unfortunately it still shows trending searches even when unchecked. Would you be able to fix so it works correctly?

**Comment by techgamers2** [comment]
can restricted mode like youtube be added just to block 18+ content coz child mode completely removes search

**Comment by yuliskov** [comment]
ok

**Comment by pjfsilva** [comment]
Any thoughts about adding another part of parental controls that is limiting time usage? 

the more advanced would be deciding what hours of the day they can use the app and the total amount of view time per day/ week.

the basic feature I think is very helpful is limiting total usage time per day. That would already be a major improvement to me. 

Older android TV versions don’t have this from the OS itself.

**Comment by Queuecumber** [comment]
Could you please also take a look at #3039 ? It currently defeats child mode

**Comment by ACR-Jeff** [comment]
> I'm waiting for some way to filter videos so that only videos from the channels I'm subscribed to on the child profile are seen. That or at least being able to only play videos from a playlist, does anyone know if it's possible?

Couldn't you just create a `Subscription Group` and add the channels you want to that Subscription group, Remove all other pinned tabs from the sidebar along with the search feature? This will leave only the channels you choose to have in the sidebar within the Subscription group?
